js/data.js

```diff
@@ -1,8 +1,40 @@
 /* exported data */
+var $photoUrl = document.querySelector('#photoUrl');
+var $picView = document.querySelector('.picView');
+
+function showPicture(event) {
+  if (event.target.value !== '') {
+    $picView.setAttribute('src', event.target.value);
+  }
+}
+
+$photoUrl.addEventListener('input', showPicture);
 
 var data = {
   view: 'entry-form',
   entries: [],
   editing: null,
   nextEntryId: 1
 };
+
+var $entryForm = document.querySelector('#entryForm');
+var entryId = 0;
+
+$entryForm.addEventListener('submit', function () {
+  event.preventDefault();
+  entryId++;
+  var entryData = {};
+  entryData.title = $entryForm.elements.title.value;
+  entryData.photoUrl = $entryForm.elements.photoUrl.value;
+  entryData.notes = $entryForm.elements.notes.value;
+  entryData.nextEntryId = entryId;
```

Comment:
You should be using `data.nextEntryId`, there is no need to create your own separate variable here.